libs/wingc/src/type_check.rs

```diff
@@ -5756,9 +5763,10 @@ mod tests {
 
 	#[test]
 	fn function_subtyping_parameter_contravariance() {
-		let void = UnsafeRef::<Type>(&Type::Void as *const Type);
-		let string = UnsafeRef::<Type>(&Type::String as *const Type);
-		let opt_string = UnsafeRef::<Type>(&Type::Optional(string) as *const Type);
+		let void = UnsafeRef::<Type>(&Type::Void);
+		let string = UnsafeRef::<Type>(&Type::String);
+		let opt_string_type = Type::Optional(string);
+		let opt_string = UnsafeRef::<Type>(&opt_string_type);
```

Comment:
ℹ️ This change also exemplefies the issue regarding memory issues: In release mode, the pointer given by `UnsafeRef::<Type>(&Type::Optional(string) as *const Type)` becomes invalid later. 

The `Type::Optional(string)` part needs to be extracted, presumably so the data lives at its expected place longer.